### ISSUE TITLE
Hardcoded slide height if no slides

### ISSUE THREAD

**Issue by krom** [FormidableLabs/nuka-carousel#105]
Use this.props.initialSlideHeight instead 100 at https://github.com/kenwheeler/nuka-carousel/blob/master/src/carousel.js#L660

If Slides added dynamically initial height will be 100 until page size changed

**Comment by kenwheeler** [comment]
want to PR that?

**Comment by Antherkiv** [comment]
Hello I facing a similar situation, I had the component as a header coupling with react-router and I fetch the image on client (universal rendering), so the component never goes well, if I put the little hack suggested by someone here, the component it's 0 until the images are loaded, I don't know if you had a good idea to handle, thanks

**Comment by ryan-roemer** [comment]
Is this still an issue on `nuka-carousel@3.1.0+` now that we've landed https://github.com/FormidableLabs/nuka-carousel/pull/291?

**Comment by ncuillery** [comment]
@ryan-roemer Yes it is (I'm using 4.1.0)

PR #111 has been closed. @stefangordon's fix was looking good to me. 🙁

I've ended up with this ugly fix:
```javascript
const FixedCarousel = ({ children, initialSlideHeight, ...otherProps}) => {
  const slides = this.props.children.length 
    ? this.props.children 
    : <div style={{ height: initialSlideHeight }} />;

  return (
    <Carousel
      initialSlideHeight={initialSlideHeight}
      {...otherProps}
    >
      {slides}
    </Carousel>
  );
}

```

**Comment by sarmeyer** [comment]
fixed in v4.3.10 after PR #412
